lib/oli_web/live_session_plugs/set_sidebar.ex

```diff
@@ -43,6 +49,30 @@ defmodule OliWeb.LiveSessionPlugs.SetSidebar do
     end
   end
 
+  defp is_same_workspace(nil, _current_url) do
+    true
+  end
+
+  defp is_same_workspace(previous_url, current_url) do
+    previous_workspace = process_url(previous_url)
+    current_workspace = process_url(current_url)
+
+    previous_workspace == current_workspace
+  end
```

Comment:
Good idea! For clarity, I think we could rename the function to something like extract_workspace instead of process_url. I'll make those changes too. Thanks, Santi!"

Reply:
In the end, I chose the long version to provide more insight.